lib/utils/log/formatter_test.go

```diff
@@ -467,3 +465,170 @@ func TestConcurrentOutput(t *testing.T) {
 		wg.Wait()
 	})
 }
+
+// allPossibleSubsets returns all combinations of subsets for the
+// provided slice, including the nil/empty set.
+func allPossibleSubsets(in []string) [][]string {
+	// include the empty set in the output
+	subsets := [][]string{nil}
+	length := len(in)
+
+	for subsetBits := 1; subsetBits < (1 << length); subsetBits++ {
+		var subset []string
+
+		for object := 0; object < length; object++ {
+			if (subsetBits>>object)&1 == 1 {
+				subset = append(subset, in[object])
+			}
+		}
+		subsets = append(subsets, subset)
+	}
+	return subsets
+}
+
+// TestExtraFields validates that the output is identical for the
+// logrus formatter and slog handler based on the configured extra
+// fields.
+func TestExtraFields(t *testing.T) {
+	// Capture a fake time that all output will use.
+	now := clockwork.NewFakeClock().Now()
+
+	// Capture the caller information to be injected into all messages.
+	pc, _, _, _ := runtime.Caller(0)
+	fs := runtime.CallersFrames([]uintptr{pc})
+	f, _ := fs.Next()
+	callerTrace := &trace.Trace{
+		Func: f.Function,
+		Path: f.File,
+		Line: f.Line,
+	}
+
+	const message = "testing 123"
+
+	// Test against every possible configured combination of allowed format fields.
+	fields := allPossibleSubsets(defaultFormatFields)
+
+	t.Run("text", func(t *testing.T) {
+		for _, configuredFields := range fields {
+			name := "not configured"
+			if len(configuredFields) > 0 {
+				name = strings.Join(configuredFields, " ")
+			}
+
+			t.Run(name, func(t *testing.T) {
+				logrusFormatter := TextFormatter{
+					ExtraFields: configuredFields,
+				}
+				// Call CheckAndSetDefaults to exercise the extra fields logic. Since
+				// FormatCaller is always overridden within CheckAndSetDefaults, it is
+				// explicitly set afterward so the caller points to our fake call site.
+				require.NoError(t, logrusFormatter.CheckAndSetDefaults())
+				logrusFormatter.FormatCaller = callerTrace.String
+
+				var slogOutput bytes.Buffer
+				var slogHandler slog.Handler = NewSlogTextHandler(&slogOutput, SlogTextHandlerConfig{ConfiguredFields: configuredFields})
+
+				entry := &logrus.Entry{
+					Data:    logrus.Fields{"animal": "llama", "vegetable": "carrot", trace.Component: "test"},
+					Time:    now,
+					Level:   logrus.DebugLevel,
+					Caller:  &f,
+					Message: message,
+				}
+
+				logrusOut, err := logrusFormatter.Format(entry)
+				require.NoError(t, err)
+
+				record := slog.Record{
+					Time:    now,
+					Message: message,
+					Level:   slog.LevelDebug,
+					PC:      pc,
+				}
+
+				record.AddAttrs(slog.String(trace.Component, "test"), slog.String("animal", "llama"), slog.String("vegetable", "carrot"))
+
+				require.NoError(t, slogHandler.Handle(context.Background(), record))
+
+				require.Equal(t, string(logrusOut), slogOutput.String())
+			})
+		}
+	})
+
+	t.Run("json", func(t *testing.T) {
```

Comment:
Looks like these two tests "text" and "json" can be merged? There isn't much except for creating `slogHandler` with either `NewSlogTextHandler` or `NewSlogJSONHandler` and some extra assertion job for json.